docs/automated-testing/test-planning.md

```diff
@@ -0,0 +1,219 @@
+    - Refine the test cases based on feedback
+
+### Examples of Test Cases using the Given-When-Then Format
+
+| User Login | (Positive Test Case) |
+| -- | -- |
+| **Title** | Verify user login with valid credentials |
+| **Preconditions** | User is on the login page |
+| **Test Steps** | **Given** the user is on the login page **When** the user enters valid credentials and clicks the login button **Then** the user should be redirected to the dashboard |
+| **Expected Result** | User is successfully logged in and redirected to the dashboard |
+
+| User Login with Invalid Credentials | (Negative Test Case) |
+| -- | -- |
+| **Title** | Verify user login with invalid credentials |
+| **Preconditions** | User is on the login page |
+| **Test Steps** | **Given** the user is on the login page **When** the user enters invalid credentials and clicks the login button **Then** the user should se an error message indicating invalid username or password |
+| **Expected Result** | Error message is displayed and the user remains on the login page |
+
+| Security Testing for Login Page | (Non-Functional Requirement) |
+| -- | -- |
+| **Title** | Verify that the login page is protected against SQL injection attacks |
+| **Preconditions** | User is on the login page |
+| **Test Steps** | **Given** the user is on the login page **When** the user enters a SQL injection string (e.g. ' OR '1'=1) in the username field and clicks the login button **Then** the system should not allow login and should display an error message |
+| **Expected Result** | The system prevents login and displays an error message |
+
+| Usability Testing for Form Validation | (Non-Functional Requirement) |
+| -- | -- |
+| **Title** | Verify that the registration form provides clear error messages for invalid inputs |
+| **Preconditions** | User is on the registration page |
+| **Test Steps** | **Given** the user is on the registration page **When** the user enters invalid data (e.g. incorrect email format) and submits the form **Then** the system should display a clear and specific error message indicating the issue |
+| **Expected Result** | Clear and specific error message is displayed |
+
+| Reliability Testing for System Uptime | (Non-Functional Requirement) |
+| -- | -- |
+| **Title** | Verify that the system maintains 99.9% uptime over a month |
+| **Preconditions** | System is monitored over a month |
+| **Test Steps** | **Given** the system is up and running **When** the system is monitored for uptime over a month **Then** the system should maintain 99.9% uptime |
+| **Expected Result** | System maintains 99.9% uptime over the monitored period |
+
+## Grouping Test Cases into Test Plans
+
+We can create different test plans, such as a full regression test plan, a smoke test plan, a functional test plan etc. and organize the test cases under these test plans.
```

Comment:
Do we want to add a link here to TDD in unit tests as an example? we have content for it...

Reply:
will add the link

Reply:
went back and looked, where was the mention of TDD that we wanted to link to the TDD content?